[Enhancement]: Highlight templates and hooks changes for community PRs

### Describe the solution you'd like

We have the useful [**highlight templates and hooks changes** workflow](https://github.com/woocommerce/woocommerce/blob/be041e0d386082644c17b59ddc406f7758302db2/.github/workflows/pr-highlight-changes.yml) but it seems not to run in the case of community PRs (ie, when we're looking at a change made via a fork of our repo).

<details>
<summary><a href="https://github.com/woocommerce/woocommerce/runs/6698227450?check_suite_focus=true">Recent example</a> and some details.</summary>
<pre>
Run ./tools/code-analyzer/bin/dev analyzer "$GITHUB_HEAD_REF" -o github
Fetching trunk... done
From https://github.com/woocommerce/woocommerce
 * branch                  trunk      -> FETCH_HEAD
 * [new branch]            trunk      -> origin/trunk
Fetching add/wc-admin-settings... !
fatal: couldn't find remote ref add/wc-admin-settings
Error: Unable to fetch add/wc-admin-settings
    at Object.error (/home/runner/work/woocommerce/woocommerce/node_modules/.pnpm/@oclif+core@1.3.4/node_modules/@oclif/core/lib/errors/index.js:29:15)
    at Analyzer.error (/home/runner/work/woocommerce/woocommerce/node_modules/.pnpm/@oclif+core@1.3.4/node_modules/@oclif/core/lib/command.js:75:23)
    at /home/runner/work/woocommerce/woocommerce/tools/code-analyzer/src/commands/analyzer/index.ts:87:32
    at fetchBranch (/home/runner/work/woocommerce/woocommerce/tools/code-analyzer/src/git.ts:39:3)
    at generatePatch (/home/runner/work/woocommerce/woocommerce/tools/code-analyzer/src/git.ts:69:13)
    at Analyzer.run (/home/runner/work/woocommerce/woocommerce/tools/code-analyzer/src/commands/analyzer/index.ts:83:37)
    at async Analyzer._run (/home/runner/work/woocommerce/woocommerce/node_modules/.pnpm/@oclif+core@1.3.4/node_modules/@oclif/core/lib/command.js:52:22)
    at async Config.runCommand (/home/runner/work/woocommerce/woocommerce/node_modules/.pnpm/@oclif+core@1.3.4/node_modules/@oclif/core/lib/config/config.js:226:25)
    at async Object.run (/home/runner/work/woocommerce/woocommerce/node_modules/.pnpm/@oclif+core@1.3.4/node_modules/@oclif/core/lib/main.js:73:5)
Error: Process completed with exit code 2.
</pre>

<p>Note the attempt to fetch the branch:</p>

<pre>
Fetching add/wc-admin-settings... !
fatal: couldn't find remote ref add/wc-admin-settings
</pre>

<p>In cases like this, we probably need to something like <code>git fetch origin pull/32728/head:add/wc-admin-settings</code> first of all (<a href="https://github.com/woocommerce/woocommerce/pull/32728">pull/32728</a> being the PR in this example). Or, perhaps it would be viable to use the GitHub command line tool to take care of these details?</p>
</details>



### Describe alternatives you've considered

Alternative is not to worry about this, for community PRs.

### Additional context

_No response_

--- a/.github/workflows/pr-highlight-changes.yml
+++ b/.github/workflows/pr-highlight-changes.yml
@@ -18,8 +18,9 @@ jobs:
               id: run
               working-directory: tools/code-analyzer
               run: |
-                  version=$(pnpm run analyzer major-minor "${{ github.head_ref || github.ref_name }}" "plugins/woocommerce/woocommerce.php" | tail -n 1)
-                  pnpm run analyzer "$GITHUB_HEAD_REF" $version -o "github"
+                  HEAD_REF=$(git rev-parse HEAD)
+                  version=$(pnpm run analyzer major-minor "$HEAD_REF" "plugins/woocommerce/woocommerce.php" | tail -n 1)
+                  pnpm run analyzer "$HEAD_REF" $version -o "github"
             - name: Print results
               id: results
               run: echo "::set-output name=results::${{ steps.run.outputs.templates }}${{ steps.run.outputs.wphooks }}${{ steps.run.outputs.schema }}${{ steps.run.outputs.database }}"

